guides/plugins/apps/administration/flow-builder/add-custom-flow-actions-from-app-system.md

```diff
@@ -0,0 +1,255 @@
+{% endhint %}
+
+## Define the flow action
+
+To create a flow action, you need to define a `<flow-action>` block within a file called `flow-action.xml`. Each `<flow-action>` represents one action and you can define an arbitrary number of actions.
+
+{% code title="Resources/flow-action.xml" %}
+```xml
+<flow-actions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/platform/trunk/src/Core/Framework/App/FlowAction/Schema/flow-action-1.0.xsd">
+    <flow-action>
+        ... # The first action
+    </flow-action>
+    <flow-action>
+        ... # The second action
+    </flow-action>
+    <flow-action>
+        ... # The third action
+    </flow-action>
+    ...
+</flow-actions>
+```
+{% endcode %}
+
+A single flow action would look like this:
+
+```xml
+<flow-action>
+    <meta>
+        <name>slack.send.message</name>
+        <label>Send slack message</label>
+        <label lang="de-DE">Send slack message</label>
+        <badge>Slack</badge>
+        <description>Slack send message description</description>
+        <description lang="de-DE">Slack send message description DE</description>
+        <url>https://hooks.slack.com/services/{id}</url>
+        <sw-icon>default-communication-speech-bubbles</sw-icon>
+        <icon>slack.png</icon>
+        <requirements>orderAware</requirements>
+        <requirements>customerAware</requirements>
+    </meta>
+    ...
+</flow-action>
+```
+
+| Key | Required | Description |
+| :--- | :--- | :--- |
+| name | yes | The technical name of your action, unique for all actions |
+| label | yes | A name to be shown for your action in the actions list or action modal title |
+| badge | no | An attached badge shown behind the label in the action modal title |
+| description | yes | Detailed information for your action |
+| sw-icon | no | An icon component name from the [icon library](https://component-library.shopware.com/icons/) |
+| icon | no | Alternatively, a path to your action icon |
+| requirements | yes | Available action triggers, read more below |
+| url | yes | External webhook location. Shopware will call this URL when the action is executed |
+
+**requirements**
+
+Requirements will decide for which trigger events you action is available.
+Example: The `checkout.order.placed` has an `orderAware` requirements - indicating that your action is allow to use be used in the `checkout.order.placed` event. It is defined using `<requirements>orderAware</requirements>` in your app action definition.
+
+### Header parameters
+
+```xml
+<flow-action>
+    <meta>
+        ...
+    </meta>
+    <headers>
+        <parameter type="string" name="content-type" value="application/json"/>
+    </headers>
+    ...
+<flow-action>
+ ```
+
+| Key | Description |
+| :--- | :--- |
+| type | Parameter type - currently only `string` supported |
+| name | The header key |
+| value | The header value |
+
+### Parameters
```

Comment:
can we add a section where we show the whole example config (with params headers etc.) and then the resulting http request that is created based on that config?